### DISCUSSION TITLE
Developer dashboard api for whitelisting domains.

### DISCUSSION THREAD

**Discussion by anand-nftically** [Web3Auth/Web3Auth#398] [Help]
Is there any way to add domain to whitelist using apis (something like dashboard apis) so we can add domain on the go. As we cannot manually add each domain for end user actions ,we have a multi tenant app.

**Comment by YZhenY** [comment]
Not at this point, could consider if there is enough interest

**Comment by himanshuchawla009** [comment]
However we don't have api to do this, but you can sign your domains in your server side code and pass those signatures in web3auth sdk without even whitelisting them on developer dashboard, here are the steps to do that:-

1. Create a project on developer dashboard.
2.  Note your client id and secret.
3.  Sign your domain programmatically. You can use following code to do signing in node js:-


```
const base64url= require("base64url");
      const keccak = require("keccak");
      const { getPublic, sign } = require("@toruslabs/eccrypto");
      // Replace it with your client id and secret
      const clientId = "BG0djgWQ8-sDz3OfdJmuQnO3XnsO8VOBhNGOSW8BgmgiNiit1374bM89r3RT0XP2n2BNkTS6A33UkQryCfkGbBs";
      const clientSecret = "f785b7a8ef9e50bd878180c4b89072d08e8c2c59840bc01ef080f5b00783264e";
      const origin = "https://tenant1.example.com"; // your tenant1 domain
      const appKeyBuf = Buffer.from(clientSecret.padStart(64, "0"), "hex");
      if (base64url.encode(getPublic(appKeyBuf)) !== clientId) throw new Error("appKey mismatch");
      const sig = await sign(appKeyBuf, Buffer.from(keccak("keccak256").update(origin).digest("hex"), "hex"));
      const finalSig = base64url.encode(sig);
      console.log("final sig", finalSig);
  })()
```

4. Pass signatures in web3auth's openlogin adapter :-

```
  const whitelistedDomains = { 
    "https://tenant1.example.com": "TENANT1_ DOMAIN_SIGNATURE", // signature that you generated above in step 3
   "https://tenant2.example.com": "TENANT2_ DOMAIN_SIGNATURE", 
}
  const adapter = new OpenloginAdapter({    adapterSettings: { originData:  whitelistedDomains }  });

```


@anand-nftically

**Reply by anand-nftically** [reply]
yes it worked, just a change we have to pass origin data inside adapterSettings : {}

`adapterSettings: {
          clientId: "{{config('constants.app.web3auth_client_id')}}",
            defaultLanguage: "en",
            dark: false, // whether to enable dark mode. defaultValue: false
          },
          originData: {
            "https://white.nftically.test" : 'MEQCIHYoDtL7L6JxZNGPSM0T3rdxXSqzQkLwrt-wuL5s3BtMAiArkxNjta0zkvC_Ep7UJsh9CgWSk2Cwe8p1veaBNNpaRw'
          }
        },`

**Reply by himanshuchawla009** [reply]
thanks for validating :), i have edited my answer.

**Comment by himanshuchawla009** [comment]
here is the updated answer for this: https://web3auth.io/docs/dashboard-setup/whitelisting#manual-whitelisting-for-advanced-cases
